src/sync/remote_mongo_collection.cpp

```diff
@@ -22,114 +22,93 @@
 namespace realm {
 namespace app {
 
-static void handle_delete_count_response(util::Optional<AppError> error,
-                                         util::Optional<bson::Bson> value,
-                                         std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+inline std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)> handle_delete_count_response(
```

Comment:
A few things, I don't think this needs to be inlined. Also, can we give this return type a shorter name? Maybe:

using ResponseHandler = std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)>;
ResponseHandler handle_delete_count_response(...)
{
    ...
}

Reply:
Also, can we put all these bare functions in an anonymous namespace?